### PR TITLE
[REVIEW] Drop support for `skiprows` and `num_rows` in `cudf.read_parquet`

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
This PR removes support for `skiprows` & `num_rows` in parquet reader. A continuation of https://github.com/rapidsai/cudf/pull/11218

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by galipremsagar** [comment]
cc: @nvdbaranec @vuule

**Review by vyasr — APPROVED** [review]

File: python/cudf/cudf/tests/test_parquet.py [hunk 1]
```diff
@@ -987,18 +933,13 @@ def L(list_size, first_val):
...
     Generate a list column based on input parameters.
 
     Args:
         gen: A callable which generates an individual leaf element based on an
             absolute index.
-        skiprows : Generate the column as if it had started at 'skiprows'
-            instead of 0. The intent here is to emulate the skiprows
-            parameter of the parquet reader.
         num_rows : Number of rows to generate.  Again, this is to emulate the
             'num_rows' parameter of the parquet reader.
```
Comment (vuule) [inline python/cudf/cudf/tests/test_parquet.py:944]: The comment here needs to be updated, or perhaps the argument needs to be changed/removed. CC @nvdbaranec

**Review by vuule — CHANGES_REQUESTED** [review]
One small change request

File: python/cudf/cudf/tests/test_parquet.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/tests/test_parquet.py]: Done 👍

**Review by vuule — APPROVED** [review]
💯

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**